packages/gafl-webapp-service/src/pages/contact/name/__tests__/update-transaction.spec.js

```diff
@@ -0,0 +1,73 @@
+import { NAME } from '../../../../uri.js'
+import updateTransaction from '../update-transaction.js'
+
+const VALID_PAYLOAD = {
+  'first-name': 'Luke',
+  'last-name': 'Skywalker'
+}
+
+const transactionHelperMock = {
+  get: jest.fn(),
+  getCurrentPermission: jest.fn(() => ({ licensee: { firstName: null, lastName: null } })),
+  setCurrentPermission: jest.fn()
+}
+
+const createRequestMock = payload => ({
+  cache: jest.fn(() => ({
+    helpers: {
+      page: {
+        get: jest.fn(() => ({
+          permissions: [
+            {
+              [NAME.page]: {
+                payload: VALID_PAYLOAD
+              }
+            }
+          ]
+        })),
+        getCurrentPermission: jest.fn(() => ({ payload: payload || { test: 'payload' } }))
+      },
+      status: {
+        get: jest.fn()
+      },
+      transaction: transactionHelperMock
+    }
+  }))
+})
+
+describe('update-transaction', () => {
+  beforeEach(jest.clearAllMocks)
+  it('gets the payload from the name page cache', async () => {
+    const mockRequest = createRequestMock(VALID_PAYLOAD)
+    await updateTransaction(mockRequest)
+    expect(mockRequest.cache.mock.results[0].value.helpers.page.getCurrentPermission).toBeCalledWith(NAME.page)
```

Comment:
This doesn't read very well. Rather than trying to fish out such a deep-lying value, in createRequestMock allow getCurrentPermission to be passed in, then you can set this up inside of the test, like this:
```
const samplePayload = {
   'first-name': 'Luke',
   'last-name': 'Skywalker'
 }
const getCurrentPermission = jest.fn(() => samplePayload)
const mockRequest = createRequestMock(samplePayload, getCurrentPermission)
await updateTransaction(mockRequest)
expect(getCurrentPermission).toHaveBeenCalledWith(NAME.page)
```